Replace horizontalBar with indexAxis: 'y'

kurkle opened the pull request:
Revisiting this one (original #7144)

Resolves: #4096, resolves: #4047

[Pen](https://codepen.io/kurkle/pen/yLeaxdL)

benmccann commented:
It might be useful to have a test where the x and y axes have different IDs to make sure that `indexAxis` still works in that case

etimberg commented on src/core/core.controller.js:
```diff
@@ -18,38 +18,62 @@ import {version} from '../../package.json';
+		const defaultId = getDefaultScaleIDFromAxis(axis, chartIndexAxis);
 		firstIDs[axis] = firstIDs[axis] || id;
-		scales[id] = mergeIf({}, [configScales[id], chartDefaults.scales[axis]]);
+		scales[id] = mergeIf({axis}, [scaleConf, chartDefaults.scales[axis], chartDefaults.scales[defaultId]]);
```
Great that we have double fallback here. I suspect there's a fair bit of code doing something like `Chart.defaults.bar.scales.y...`

kurkle commented:
> It might be useful to have a test where the x and y axes have different IDs to make sure that `indexAxis` still works in that case

Good catch, it didn't actually work. Fixed and added a mixed test.

benmccann commented:
I think this closes https://github.com/chartjs/Chart.js/issues/4047 as well

kurkle commented:
> scatter next? ;-)

Not sure what you mean, this works for scatter too: https://codepen.io/kurkle/pen/JjGbVeG?editors=1010

benmccann commented:
I meant maybe we should find figure out how to delete it next
